### PR TITLE
MGMT-18313: Replace golang base image as it is based on Centos Linux 7

### PR DESCRIPTION
**Pull request by danmanor**
## This PR 

* Replace Centos Linux 7 based golang images with ubi9 based golang images
* Replaces deprecated lint package

This PR will be later followed by a PR to apply the new build arg in openshift/release

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [x] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: Dockerfile.assisted-service [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+
+ADD . /app
+WORKDIR /app
+
+# Licenses
+RUN go install github.com/google/go-licenses@v1.6.0
+RUN ${HOME}/go/bin/go-licenses save --save_path /tmp/licenses ./...
 
 # Build binaries
-FROM quay.io/centos/centos:$BASE_TAG as builder
```
Comment (pastequo) [inline Dockerfile.assisted-service:4]: I think this build-arg is used in another repo..
cc @carbonin

File: Dockerfile.assisted-service [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
+FROM quay.io/centos/centos:stream$RHEL_VERSION as builder
```
Comment (adriengentil) [inline Dockerfile.assisted-service:13]: why we don't build from the UBI?

File: Dockerfile.assisted-service (hunk 1 above)
Comment (carbonin) [reply Dockerfile.assisted-service:4]: Looks like this was already updated in openshift/release https://github.com/openshift/release/blob/9d8442afb6e832c1c722bccaef20f7c0950d227b/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml#L131-L134

But please to remember to consider the downstream impacts of this (if any)

File: Dockerfile.assisted-service (hunk 2 above)
Comment (eifrach) [reply Dockerfile.assisted-service:13]: related to the subscription manager needed on UBI images.

**Review by adriengentil — APPROVED** [review]

**Pull request merged**